### ISSUE TITLE
Split file doesn't account encryption overhead for 2GiB parts

### ISSUE THREAD

**Issue by R4to0** [divyam234/teldrive#287]
tl;dr encryption gives a overhead of 511922 bytes when uploading large files with 2GiB split through UI, hanging uploading process. Disable encryption or turn down split size in webui.

----

I've noticed this when I tried to use its max split file size of 2GiB. Without encryption I can upload files larger than that just fine and splitter did its job by making parts of exactly 2097152000 bytes (2 * 1000 * 1024 * 1024), however switching encryption to ON makes uploader hangs for five minutes before giving a feedback in the terminal.

![346052950-25d21fdd-c3c1-4956-9885-2079723817e9](https://github.com/divyam234/teldrive/assets/3608184/97b8f760-db3c-45bd-b75c-74f79d515749)

```
05/07/2024 04:23 AM     DEBUG  uploading chunk {"fileName": "test2GiB", "partName": "test2GiB", "bot": "[redacted]", "botNo": 3, "chunkNo": 1, "partSize": 2097152000}
05/07/2024 04:28 AM     DEBUG  upload failed   {"fileName": "test2GiB", "partName": "test2GiB", "chunkNo": 1}
05/07/2024 04:28 AM     ERROR  callback: upload part: send upload part 0 RPC: retry middleware skip: invoke pool: rpcDoRequest: rpc error code 400: FILE_PARTS_INVALID
05/07/2024 04:28 AM     INFO           {"status": 500, "method": "POST", "path": "/api/uploads/1babc34ec4896ed3e2ad17e71f59c45e", "query": "partName=test2GiB&fileName=test2GiB&partNo=1&encrypted=true&channelId=0", "ip": "127.0.0.1", "user-agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0", "latency": 301.153424, "time": "2024-07-05T07:28:18Z"}
```

The above repeats in a loop until I cancel uploading a file.

Having done a lot of trial and error with generating a dummy file, I found that with encryption we can only have a max size of exact 2096640078 bytes, one more byte and fails with issue above.

![346052990-8c4f04b5-b158-4b58-83c5-27f7cc5689ae](https://github.com/divyam234/teldrive/assets/3608184/c1e3253c-2a1e-4d69-bb9c-29d1ca40858d)

I've downloaded the (encrypted) file part from my telegram client just to check its size after encryption and it's **2097152000** bytes now, giving in a overhead of 511922 bytes on top of the original file size when encrypted. In rclone I have to set `chunk_size` to `2096640078b` to be able to upload files around ~2.5GiB.

**Comment by divyam234** [comment]
@R4to0  2GB split size is not recommended set it to 500MB to get max upload speed also this is not a issue you have to decrease chunk_size for encryption.

**Comment by R4to0** [comment]
Yes I'm aware of the recommended 500MB and the use of your rclone fork instead of the web UI, I was just curious to try this. Thanks for checking it.
